### PR TITLE
clean up service provider

### LINKED ISSUE #32: Clean up the service provider
The service provider contains too much logic, consider separating the route binding

### PR DESCRIPTION
**Pull request by johnnyfekete**
Fixes #32 

I added the 4 route resolvers to the folder, as it was discussed in the issue; each one with a static method: `bind`

and the `NebulaServiceProvider` calls these methods in its `boot` method.

### CODE REVIEW

**Comment by Larsklopstra** [comment]
Could you add an abstract class `Resolver` with a `bind` method?

File: src/Http/RouteResolvers/DashboardResolver.php [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+     *
+     * @return void
+     * @throws BindingResolutionException
+     * @throws HttpException
+     * @throws NotFoundHttpException
+     */
+    public static function bind(): void
+    {
+        Route::bind('dashboard', function ($value) {
+            $dashboards = config('nebula.dashboards');
```
Comment (ryangjchandler) [inline src/Http/RouteResolvers/DashboardResolver.php]: This should be using `Nebula::availableDashboards()` now.

File: src/Http/RouteResolvers/ResourceResolver.php [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+     *
+     * @return void
+     * @throws BindingResolutionException
+     * @throws HttpException
+     * @throws NotFoundHttpException
+     */
+    public static function bind(): void
+    {
+        Route::bind('resource', function ($value) {
+            $resources = config('nebula.resources');
```
Comment (ryangjchandler) [inline src/Http/RouteResolvers/ResourceResolver.php]: This should be using `Nebula::availableResources()`.

**Comment by Larsklopstra** [comment]
@ryangjchandler could you check this now?

**Comment by Larsklopstra** [comment]
Looks good to me

**Pull request merged**